_posts/2025-03-10-data-wants-to-be-free-japanese.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: post
+lang: ja-JP
+title: "データは自由になりたい：Apache Arrowでデータ交換は高速"
+description: ""
+date: "2025-02-28 00:00:00"
+author: David Li, Ian Cook, Matt Topol
+categories: [application]
+image:
+  path: /img/arrow-result-transfer/part-1-share-image.png
+  height: 1200
+  width: 705
+translations:
+  - language: 原文（English）
+    post_id: 2025-02-28-data-wants-to-be-free
+---
+
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+<style>
+.a-header {
+  color: #984EA3;
+  font-weight: bold;
+}
+.a-data {
+  color: #377EB8;
+  font-weight: bold;
+}
+.a-length {
+  color: #FF7F00;
+  font-weight: bold;
+}
+.a-padding {
+  color: #E41A1C;
+  font-weight: bold;
+}
+</style>
+
+_この記事はデータベースとクエリーエンジン間のデータ交換フォーマットとしてなぜArrowが使われているのかという謎を解くシリーズの２記事目です。_
+
+{% include arrow_result_transfer_series_japanese.md %}
+
+データ技術者として、データはよく「人質に取られた」ことをわかっています。
+データをもらい次第データを使わずに、まず時間を掛けなければいけません。
+非効率的で雑然ＣＳＶファイルを整理する時間を掛けたり、
+型落ちのクエリエンジンが数GBのデータに苦労するのを待つ時間を掛けたり、
+データがソケットを介して受信するのを待つ時間を掛けたり。
+今回はこの三目の問題を注目します。
+マルチギガビットネットワークの時代に、そもそもこの問題がまだ起こっているのはどうしてでしょうか？
+間違いなく、この問題はぜったいまだ起こっています。
+[Mark RaasveldtとHannes Mühleisenの２０１７年の論文](https://doi.org/10.14778/3115404.3115408)[^freepdf]によって、１０秒しかかからないはずのデータセットを送受信はあるデータシステムが**１０分**以上をかかります[^ten]。
+
+[^freepdf]: [VLDB](https://www.vldb.org/pvldb/vol10/p1022-muehleisen.pdf)から論文を無料でダウンロードできます。
+[^ten]: 論文のFigure 1に、ベースラインとしてnetcatは１０秒でＣＳＶファイルを送信にたいして、HiveとMongoDBは６００秒以上がかかるのを示します。もちろん、ＣＳＶは解析されてないので、この比較は完全に平等のではありません。でも問題の規模を把握できます。
+
+どうして必要より６０倍以上の時間がかかるでしょうか？
+[この前に論じていた通り、ツールはデータシリアライズのオーバーヘッドに悩まされています。]({% link _posts/2025-01-10-arrow-result-transfer.md %})
+でもArrowは手伝えます。
+それでもっと具体的にします：データシリアライズフォーマットの影響を示すために、PostgreSQLとArrowは同じデータをどうやってエンコードするのを比較します。
```

Comment:
In general, we can use "。" for ":" in Japanese. In general, ":" isn't used in Japanese.

```suggestion
それではもっと具体的な話をしましょう。データシリアライズフォーマットの影響を示すために、PostgreSQLとArrowが同じデータをどうやってエンコードするのかを比較しましょう。
```